src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/action/index/TransportIndexTransformAction.kt

```diff
@@ -0,0 +1,141 @@
+import com.amazon.opendistroforelasticsearch.indexmanagement.IndexManagementPlugin.Companion.INDEX_MANAGEMENT_INDEX
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.action.get.GetTransformAction
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.action.get.GetTransformRequest
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.action.get.GetTransformResponse
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.model.Transform
+import com.amazon.opendistroforelasticsearch.indexmanagement.util.IndexUtils
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.ElasticsearchStatusException
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.DocWriteRequest
+import org.elasticsearch.action.index.IndexRequest
+import org.elasticsearch.action.index.IndexResponse
+import org.elasticsearch.action.support.ActionFilters
+import org.elasticsearch.action.support.HandledTransportAction
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.tasks.Task
+import org.elasticsearch.transport.TransportService
+import java.lang.Exception
+
+class TransportIndexTransformAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters,
+    val indexManagementIndices: IndexManagementIndices,
+    val clusterService: ClusterService
+) : HandledTransportAction<IndexTransformRequest, IndexTransformResponse>(
+        IndexTransformAction.NAME, transportService, actionFilters, ::IndexTransformRequest
+) {
+
+    private val log = LogManager.getLogger(javaClass)
+
+    override fun doExecute(task: Task, request: IndexTransformRequest, listener: ActionListener<IndexTransformResponse>) {
+        IndexTransformHandler(client, listener, request).start()
+    }
+
+    inner class IndexTransformHandler(
+        private val client: Client,
+        private val actionListener: ActionListener<IndexTransformResponse>,
+        private val request: IndexTransformRequest
+    ) {
+
+        fun start() {
+            indexManagementIndices.checkAndUpdateIMConfigIndex(ActionListener.wrap(::onCreateTransformResponse, actionListener::onFailure))
+        }
+
+        private fun onCreateTransformResponse(response: AcknowledgedResponse) {
+            if (response.isAcknowledged) {
+                log.info("Successfully created or updated $INDEX_MANAGEMENT_INDEX with newest transform.")
+                if (request.opType() == DocWriteRequest.OpType.CREATE) {
+                    putTransform()
+                } else {
+                    getTransform()
```

Comment:
can you rename this to `updateTransform()`

Reply:
Looks like you changed `putTransform` -> `updateTransform`. Sorry if I wasn't clear earlier, I meant changing `getTransform` -> `updateTransform` and not changing `putTransform`

even though the method is called `getTransform` its acutally fetching the transform and then writing to the config, hence the suggestion to rename